stl/inc/atomic

```diff
@@ -641,9 +641,23 @@ struct _Atomic_storage<_Ty, 8> { // lock-free using 8-byte intrinsics
 
 #ifdef _M_IX86
     void store(const _Ty _Value, const memory_order _Order = memory_order_seq_cst) noexcept {
-        // store with (effectively) sequential consistency

```

Comment:
Alternately, just delete the entire `_M_IX86` block, I think the other version works for x86 now.

Reply:
Then I need to redefine `_ISO_VOLATILE_STORE64` for x86, which is currently the same as x64, that is empty.
Should I redefine `_ISO_VOLATILE_STORE64` ?

Reply:
In fact, I see that `__iso_volatile_loadN` / `__iso_volatile_storeN` currently exists for every CPU, so there should not be a point for `_ISO_VLOATILE_STOREN` macro. But I'm not sure whether replacing everything with direct builtins is the right thing to do.

Reply:
The `_ISO_VOLATILE_STOREN` macros are from before x86 had all of them; you could probably eliminate those too if you feel like.

Reply:
(It would be better to eliminate more macros but I understand if that's a bigger change than you're interested in attempting right now)

Reply:
I've eliminated `_ISO_VOLATILE_STOREN` / _ISO_VOLATILE_LOADN`, and used builtins everywhere. Now I have ICE.

Reply:
Yes, now it is in the right branch, sorry for confusion.
ICE seem to be caused by `__iso_volatile_load64`

Reply:
I meant `__iso_volatile_store64` of course.

Reply:
Another issue with replacing macros with intrinsic functions.
There's no dumcumentation on x86 / x64 versions, but ARM version is documented [as follows](https://docs.microsoft.com/en-us/cpp/intrinsics/arm-intrinsics?view=vs-2019#IsoVolatileLoadStore): 

> The compiler can't remove, synthetize, or change the relative order of these operations, but it doesn't generate implicit hardware memory barriers. 

So there is implicit compiler barrier. As I understand, implementation with macro that directly uses pointer indirection does not implement compiler barrier. 

It may mean that relaxed order for intrinsics is not that much relaxed, as it was before.

Reply:
>So there is implicit compiler barrier. 

No, the compiler won't reorder the volatile things but it can reorder *other* operations around the volatile things. That's fine for relaxed atomics but not for anything stronger.

Reply:
JMears proposed a workaround for  DevCom-986061 to use inline assembly to reimplement x86 `__iso_volatile_store`

Reply:
Resolved to use load intrinsics and wait for DevCom-986061 resolution to arrive